gnosis/gp-swap-ui
# Swap: Calculate Tip

Part of #35 

Another go at computing tip on trades. This approach is the least intrusive/most readble imo

Key takeaways: 
1. extends the Trade object simply as an object and inputs the relevant `inputAmount` / `outputAmount`s that make sense 
2. pulls feeInformation and calculates fee on `inputAmout`

### Testing
1. [RINKEBY] - PRaul or locally, append: `?inputCurrency=ETH&outputCurrency=0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984&exactAmount=1`
2. input amounts, both sides, check amounts and compare to `app.uniswap.org` if u need
3. open console, paste: 
```js
addFee({
    token: 'ETH',
    fee: { minimalFee: '10', feeRatio: 10, expirationDate: new Date(Date.now() + 300000).toISOString() },
    chainId: '4'
})
```
4. see the changes and repeat

### Note
Inside `custom/state/swap/hooks.ts` there is my custom `useDerivedSwapInfo` hook which really only changes lines: `53-67`:
```ts
const feeInformation = useFee({
    token: inputCurrencyId,
    chainId
  })

  const bestTradeExactIn = useTradeExactInWithFee({
    parsedAmount,
    outputCurrency,
    feeInformation
  })
  const bestTradeExactOut = useTradeExactOutWithFee({
    parsedAmount,
    inputCurrency,
    feeInformation
  })
```

^ those functions are the ones inside `extension.ts`
